### PR TITLE
[dagster-azure] Add support for DefaultAzureCredential for adls2_resource

### PR DESCRIPTION
**Pull request by mpicard**
### Summary & Motivation

This introduces an additional, optional resource configuration for the adls2 resource that uses `azure.identity.aio.DefaultAzureCredential` as the credential for authenticating in many different environments, see [here](https://learn.microsoft.com/en-us/python/api/azure-identity/azure.identity.defaultazurecredential?view=azure-python) for more info. 

The config takes a Permissive approach to pass arbitrary keyword arguments to the class to configure it.

### How I Tested These Changes

This code is unit tested and it was also manually tested in a separate test repository and tested in an Azure AKS environment.

### CODE REVIEW

**Comment by mpicard** [comment]
@jamiedemaria

**Comment by mpicard** [comment]
Hey @jamiedemaria can this one get merged soon as well? I have been waiting to refactor this

**Comment by jamiedemaria** [comment]
@mpicard  sorry about that! this got totally lost. reviewing now

File: python_modules/libraries/dagster-azure/dagster_azure/adls2/resources.py [hunk 1]
```diff
@@ -48,8 +56,7 @@ def my_job():
     `required_resource_keys`, or it will not be initialized for the execution of their compute
     functions.
 
-    You may pass credentials to this resource using either a SAS token or a key, using
-    environment variables if desired:
+    You may pass credentials to this resource using either a SAS token, a key or by
```
Comment (jamiedemaria) [inline python_modules/libraries/dagster-azure/dagster_azure/adls2/resources.py:59]: is this sentence incomplete?

**Review by jamiedemaria — APPROVED** [review]
CI is passing and I just have the one comment!

**Comment by mpicard** [comment]
Ah very good catch, I have pushed the rest of the sentence

File: python_modules/libraries/dagster-azure/dagster_azure/adls2/resources.py (hunk 1 above)
Comment (mpicard) [reply python_modules/libraries/dagster-azure/dagster_azure/adls2/resources.py]: Ah very good catch, I have pushed the rest of the sentence

**Pull request merged**